core/config/project_settings.cpp

```diff
@@ -474,12 +474,24 @@ bool ProjectSettings::_load_resource_pack(const String &p_pack, bool p_replace_f
 		return false;
 	}
 
+	if (p_pack == "res://") {
+		// Loading the resource directory as a pack source is reserved for internal use only.
+		return false;
+	}
+
 	bool ok = PackedData::get_singleton()->add_pack(p_pack, p_replace_files, p_offset) == OK;
 
 	if (!ok) {
 		return false;
 	}
 
+	if (!using_datapack && !OS::get_singleton()->get_resource_dir().is_empty()) {
+		// Add the project's resource file system to PackedData so directory access keeps working when
+		// the game is running without a main pack, like in the editor or on Android.
+		PackedData::get_singleton()->add_pack_source(memnew(PackedSourceDirectory));
+		PackedData::get_singleton()->add_pack("res://", !p_replace_files, 0);
+	}
```

Comment:
Should it go before `PackedData::get_singleton()->add_pack` and always have `replace_files = false` when loading additional packs? Since loaded, resource pack should override files in the folder (like it will do with main pack).

Probably something like:
- Skip entirely when loading main pack (might need additional wrapper method with extra argument set for main pack).
- When loading additional pack, do it before loading pack (as if it is main pack).

```c++
// Exposed wrapper:
bool ProjectSettings::load_resource_pack(const String &p_pack, bool p_replace_files, int p_offset) {
	return _load_resource_pack(const String &p_pack, bool p_replace_files, int p_offset, false)
}

// Implementation:
bool ProjectSettings::_load_resource_pack(const String &p_pack, bool p_replace_files, int p_offset, bool p_main_pack) {

...

	if (!p_main_pack && !using_datapack && !OS::get_singleton()->get_resource_dir().is_empty()) {
		// Add the project's resource file system to PackedData so directory access keeps working when
		// the game is running without a main pack, like in the editor or on Android.
		PackedData::get_singleton()->add_pack_source(memnew(PackedSourceDirectory));
		PackedData::get_singleton()->add_pack("res://", false, 0);
	}
	bool ok = PackedData::get_singleton()->add_pack(p_pack, p_replace_files, p_offset) == OK;
...

}
```